org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/mapping/DoEntityMappings.java

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (c) 2010-2021 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+package org.eclipse.scout.rt.dataobject.mapping;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.eclipse.scout.rt.dataobject.DoNode;
+import org.eclipse.scout.rt.dataobject.IDoEntity;
+import org.eclipse.scout.rt.dataobject.IDoEntityContribution;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.exception.PlatformException;
+import org.eclipse.scout.rt.platform.holders.IHolder;
+import org.eclipse.scout.rt.platform.util.BooleanUtility;
+
+/**
+ * DO entity mappings are usually initialized via a subclass of {@link AbstractDoEntityMapper}.
+ */
+public class DoEntityMappings<DO_ENTITY extends IDoEntity, PEER> {
+
+  // Modifications to this list are not permitted once the instance of this class is created and initialized,
+  // otherwise ConcurrentModificationException can occur when to/fromDo is used concurrently with calls to the with methods.
+  protected final List<IDoEntityMapping<DO_ENTITY, PEER>> m_mappings = new ArrayList<>();
```

Comment:
AbstractDoEntityMapper could "seal" this mapping after init:

public AbstractDoEntityMapper() {
  initMappings(m_mappings);
  extendMappings(m_mappings);
	m_mappings.seal();  // block further modifications
}

Reply:
No sealing for now, access via AbstractDoEntityMapper subclass is restricted, change of existing mappings is not supported.